docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md

```diff
@@ -0,0 +1,228 @@
+
+## 3. Set up Intel DCAP Quote Provider Library (QPL):
+### 3.1 Install Intel DCAP Quote Provider by using its Debian package
+This [link](https://download.01.org/intel-sgx/sgx-dcap/1.6/linux/distro/ubuntuServer18.04/debian_pkgs/libs/libsgx-dcap-default-qpl/) lists all different version of libsgx-dcap-default-qpl, please download the most recent version that matches your OS version. For Ubuntu 18.04 (code name [Bionic Beaver](https://wiki.ubuntu.com/BionicBeaver)), please download the version libsgx-dcap-default-qpl_{version}-bionic1_amd64.deb. For Ubuntu 16.04 (code name [Xenial Xerus](https://wiki.ubuntu.com/XenialXerus)), please download the version libsgx-dcap-default-qpl_{version}-xenial1_amd64.deb.
+
+In this document, we use libsgx-dcap-default-qpl_1.6.100.2-bionic1_amd64.deb as an example, run the command blow to download the package
+```bash
+cd ~
+wget https://download.01.org/intel-sgx/sgx-dcap/1.6/linux/distro/ubuntuServer18.04/debian_pkgs/libs/libsgx-dcap-default-qpl/libsgx-dcap-default-qpl_1.6.100.2-bionic1_amd64.deb
+```
+
+Then install the package
+```bash
+sudo dpkg -i libsgx-dcap-default-qpl_1.6.100.2-bionic1_amd64.deb
+```
+
+### 3.2 Create a soft link
+OE expects the file name of the qpl to be libdcap_quoteprov.so. But the intel default qpl creates installed libdcap_quoteprov.so.1 and libdcap_quoteprov.so.1.6.100.2.  libdcap_quoteprov.so.1 is a soft link to libdcap_quoteprov.so.1.6.100.2. To allow OE works properly, we need to create the other soft link called libdcap_quoteprov.s linking to libdcap_quoteprov.so.1.6.100.2
+
+Check where those files are installed.
+```bash
+dpkg --listfiles libsgx-dcap-default-qpl
+```
+
+In most cases, it should be in /usr/lib/x86_64-linux-gnu/
+
+Use /usr/lib/x86_64-linux-gnu/ as an example.
+```bash
+cd /usr/lib/x86_64-linux-gnu/
+sudo ln -s libdcap_quoteprov.so.1.6.100.2 libdcap_quoteprov.so
+```
+
+NOTES TO USERS WHO HAVE ALREADY INSTALLED AZURE DCAP CLIENT:
+
+If you have Azure DCAP client installed before trying these instructions, please make sure Azure one is renamed to something else.
+
+To check if you have it installed, run the following command.
+```bash
+dpkg --list | grep az-dcap-client
+```
+
+If you don't have Azure DCAP client installed previously, please skip the content below.
+
+In most cases the Azure version of libdcap_quoteprov.so is located in /usr/lib. Check your path before changing. Here we use /usr/lib as an example.
+
+```bash
+sudo mv /usr/lib/libdcap_quoteprov.so /usr/lib/libdcap_quoteprov.so.azure
+```
+
+Otherwise the Azure one might still get used because $PATH might have /usr/lib before the path /usr/lib/x86_64-linux-gnu with the Intel version.
+
+### 3.3 Configure the qpl
+Edit the file /etc/sgx_default_qcnl.conf. To accept insecure HTTPS cert, set the option USE_SECURE_CERT to FALSE as we will use a local caching service which doesn't have a secure cert.
```

Comment:
Can you say more about this?   This sounds insecure, is there no secure option available?

Reply:
It is being called "insecure" because the service will only have a self-signed cert. If we want to make it a "secure" cert, we need to go through the process to have our cert registered, approved, and signed by a public CA.  

In our settings, the PCCS service will only sit in the local and only accepts local requests. So there is no security problem. Deploying a PCCS for an organization to serve the people in that org would surely need a CA-signed cert. But this document is to help individual developer set up his or her own development and test environment.  

A CA-signed/self-signed cert might be a better word than "secure/insecure" cert. But that's how exactly Intel PCCS config file describe it. We may keep that wording. But I will add some explanation to the docs.

Reply:
Does the service run in an enclave or not?   If not, what happens if it is compromised?

Reply:
The local PCCS service doesn't run in an enclave. Azure PCCS also doesn't run in an enclave. Those services are not a part of the TCB, which means they don't have to be trusted by design. An individual developer doesn't need to trust that local PCCS just like the customers who are using Azure PCCS today don't need to trust Azure PCCS (and any other parts of Azure, except their own enclave).

The cert mentioned in PCCS config is just for a regular TLS handshaking, just like what happened when we browse any HTTPS websites. That cert itself has nothing to do with attestation process. It has no relationship with the certs (e.g., Provisioning Certification Key certs (PCK certs)) that being used in the attestation.

If a local PCCS service is compromised, then the generation of remote attestation report (quote) will fail. So the security of enclave won't be compromised. Imagine a compromised local PCCS, then it can alter any bits in the Provisioning Certification Key certs (PCK certs) then returns the altered PCK certs to the enclaves. Public PCK is included in the PCK certs, which are signed by Intel Intermediate CA and then Intel Root CA in chain. Only enclaves on that specific machine has access to the private PCK which is derived from some information fused in the processor. So the enclave can verify the integrity of the PCK certs returned by PCCS, which means it is impossible to use a forged PCK cert to generate a quote.

Reply:
Added some notes for that topic in the doc.